### PR TITLE
feat: init for token supply server

### PR DESCRIPTION
**Pull request by joshuef**
This gives us an endpoint to supply to market cap sites for existing supply levels

### Summary generated by Reviewpad on 23 Jan 24 08:50 UTC
This pull request includes the following changes:

1. The `Cargo.toml` file has been modified to add the new crate "token_supplies" as a member of the workspace. It also includes changes to package metadata and the addition of dependencies.

2. A new file called `README.md` has been added in the `token_supplies` directory. It describes a server that caches and provides supply levels for the `emaid` and `maid` tokens. The server exposes two endpoints: `http://<address>:3030/maid` and `http://<address>:3030/emaid`, which return the current supply levels for the respective tokens.

3. The file `main.rs` has been modified to add imports for various dependencies, define new structs and functions, and modify the `main()` function. These changes add functionality to fetch data from an API, write it to a file, and serve the data in response to requests.

4. The `Cargo.lock` file has been updated to include several new packages and their dependencies.

Please review the details provided in the individual summaries for more information on each file's changes. Let me know if you have any specific questions.

### CODE REVIEW

File: token_supplies/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+#[tokio::main]
+async fn main() {
+    tokio::spawn(async {
+        scheduled_fetch().await;
+    });
+
+    let maid_supply = warp::path!("maid").map(|| match read_from_file() {
+        Ok(data) => format!("{}", data.maid_supply),
+        Err(_) => "Error reading data".to_string(),
```
Comment (grumbach) [inline token_supplies/src/main.rs]:
```suggestion
        Err(e) => format!("Error reading data: {e}"),
```

Comment (grumbach) [reply token_supplies/src/main.rs]: Nothing blocking, but this could help us know what happened if there's a failure.

**Review by grumbach — APPROVED** [review]
Small comment, no blockers, LGTM

**Comment by joshuef** [comment]
updated for your comment there @grumbach 🙇

**Pull request merged**